tests/state/_common/reducers-test.js

```diff
@@ -0,0 +1,1227 @@
+						k1: {
+							data: {
+								name: 'key prop in keys',
+								actionKey: 'ak1',
+							},
+						},
+						k2: {
+							data: {
+								name: 'key prop not in keys',
+								actionKey: 'nak2',
+							},
+						},
+						k3: {
+							data: {
+								name: 'multi key prop in keys',
+								actionKeys: ['ak3.1', 'ak3.2'],
+							},
+						},
+						k4: {
+							data: {
+								name: 'multi key prop not in keys',
+								actionKeys: ['ak4.1', 'ak4.2'],
+							},
+						},
+					},
+				},
+				action: {
+					dataType: 'actions',
+					keys: ['ak1', 'ak3.2'],
+				},
+				expectedResult: {
+					editedByKey: {
+						k1: {
+							data: {
+								name: 'key prop in keys',
+								actionKey: null,
+							},
+						},
+						k2: {
+							data: {
+								name: 'key prop not in keys',
+								actionKey: 'nak2',
+							},
+						},
+						k3: {
+							data: {
+								name: 'multi key prop in keys',
+								actionKeys: ['ak3.1'],
+							},
+						},
+						k4: {
+							data: {
+								name: 'multi key prop not in keys',
+								actionKeys: ['ak4.1', 'ak4.2'],
+							},
+						},
+					},
+				},
+			},
+		];
+
+		tests.forEach((test) => {
+			assert.deepStrictEqual(
+				reducers.removeEditedPropertyValues(test.state, test.action),
+				test.expectedResult
+			);
+		});
+	});
+
+	describe('setActive', function () {
+		const tests = [
+			{
+				name: 'test',
+				state: {
+					activeKey: 'k1',
+					activeKeys: ['ks1', 'ks2'],
+				},
+				action: {
+					key: 'ak',
+				},
+				expectedResult: {
+					activeKey: 'ak',
+					activeKeys: null,
+				},
+			},
+		];
+		tests.forEach((test) => {
+			assert.deepStrictEqual(
+				reducers.setActive(test.state, test.action),
+				test.expectedResult
+			);
+		});
+	});
+
+	describe('setActiveMultiple', function () {
+		const tests = [
+			{
+				name: 'setActiveMultiple',
+				state: {
+					activeKey: 'k1',
+					activeKeys: ['ks1', 'ks2'],
+				},
+				action: {
+					keys: ['aks1', 'aks2'],
+				},
+				expectedResult: {
+					activeKey: null,
+					activeKeys: ['aks1', 'aks2'],
+				},
+			},
+		];
+
+		tests.forEach((test) => {
+			assert.deepStrictEqual(
+				reducers.setActiveMultiple(test.state, test.action),
+				test.expectedResult
+			);
+		});
+	});
+
+	describe('updateEdited', function () {
+		const tests = [
+			{
+				name: 'empty data',
+				state: {
+					editedByKey: {
+						k1: {data: {p: 1, ep: 2}},
+						k2: {data: {p: 2}},
+					},
+				},
+				action: {
+					data: [],
+				},
+				expectedResult: {
+					editedByKey: {
+						k1: {data: {p: 1, ep: 2}},
+						k2: {data: {p: 2}},
+					},
+				},
+			},
+			{
+				name: 'some data',
+				state: {
+					editedByKey: {
+						k1: {data: {p: 1, ep: 2}},
+						k2: {data: {p: 2}},
+					},
+				},
+				action: {
+					data: [
+						{key: 'k1', data: {p: 11}},
+						{key: 'k3', data: {p: 33}},
+					],
+				},
+				expectedResult: {
+					editedByKey: {
+						k1: {data: {p: 11, ep: 2}},
```

Comment:
Missing key property?

Reply:
Is that a question? I saw that code I tested for the first time :) Should I add it then?

Reply:
code merges just data: https://github.com/gisat-panther/ptr-state/blob/053dc243def2fd9a33a50679aadfa821fa855c8d/src/state/_common/reducers.js#L342

Reply:
It confused me that key is missing in state data. However, it seems OK then